src/core/document.js

```diff
@@ -599,8 +611,21 @@ class Page {
         ) {
           annotationsData.push(annotation.data);
         }
+
+        if (intentAny || (intentDisplay && annotation.viewable)) {
+          textContentPromises.push(
+            annotation.extractTextContent(partialEvaluator, task).then(text => {
+              if (text) {
+                annotation.data.text = text;
```

Comment:
Would it work, and if so make sense, to instead have the `extractTextContent`-method itself assign the `data.text`-property when a non-empty string exists?